src/datreant/core/tests/test_util.py

```diff
@@ -0,0 +1,36 @@
+import mock
+import os
+import pytest
+
+import datreant.core as dtr
+
+
+class TestMakedirs(object):
+    def test_makedirs(self, tmpdir):
+        with tmpdir.as_cwd():
+            dtr.util.makedirs('this/should/exist')
+            assert os.path.exists('this/should/exist')
+            assert os.path.isdir('this/should/exist')
+
+    def test_makedirs_exists(self, tmpdir):
+        # try and make a dir twice
+        with tmpdir.as_cwd():
+            os.mkdir('this/')
+            dtr.util.makedirs('this/')
+            assert os.path.exists('this/')
+            assert os.path.isdir('this/')
+
+    def test_makedirs_error_catch(self, tmpdir):
+        # mock a disk full error
+        # and make sure it gets propagated through properly
+        with tmpdir.as_cwd():
+            with mock.patch('os.makedirs') as mp:
+                mp.side_effect = OSError(os.errno.ENOSPC, 'Mock - disk full')
+                with pytest.raises(OSError):
```

Comment:
This should simplify it

```python
with pytest.raises(OSError) as error:
     dtr.util.makedirs('this/should/fail')
assert error.errno = os.errno.ENOSPC
```

See: http://doc.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions